react-components/src/utilities/instanceIds/typeGuards.test.ts

```diff
@@ -0,0 +1,90 @@
+import { describe, it, expect } from 'vitest';
+import {
+  isIdEither,
+  isExternalId,
+  isInternalId,
+  isDmsInstance,
+  type AnnotationAssetRef
+} from '../../../src/utilities/instanceIds/typeGuards';
+import { type IdEither } from '@cognite/sdk';
+import { type DmsUniqueIdentifier } from '../../../src/data-providers';
+import { type InstanceReference } from '../../../src/utilities/instanceIds/types';
+
+describe('typeGuards', () => {
+  describe(isIdEither.name, () => {
+    it('should return true for an ExternalId', () => {
+      const instance: InstanceReference = { externalId: 'external-id' };
+      expect(isIdEither(instance)).toBe(true);
+    });
+
+    it('should return true for an InternalId', () => {
+      const instance: InstanceReference = { id: 123 };
+      expect(isIdEither(instance)).toBe(true);
+    });
+
+    it('should return false for an invalid object', () => {
+      const instance = { someKey: 'value' } as const as AnnotationAssetRef;
+      expect(isIdEither(instance)).toBe(false);
```

Comment:
Not sure that we need to account for cases where the input violates the type requirements.. But fine, we can keep it :sweat_smile: